packages/block-editor/src/store/actions.js

```diff
@@ -914,10 +914,26 @@ export function* removeBlocks( clientIds, selectPrevious = true ) {
 		clientIds,
 	};
 
+	if ( previousBlockId ) {
+		return [ previousBlockId ];
+	}
+
+	const { __unstableHasCustomAppender } = yield controls.select(
+		blockEditorStoreName,
+		'getSettings'
+	);
+
+	// If there's an custom appender, don't insert default block.
+	// We have to remember to manually move the focus elsewhere to
+	// prevent if from being lost though.
+	if ( __unstableHasCustomAppender ) {
+		return;
+	}
```

Comment:
I think this bit of code could be moved to `ensureDefaultBlock`, as that would also cover `replaceBlocks` (which also calls `ensureDefaultBlock`).

Reply:
Oh right, I forgot about that.